src/auth/src/credentials/idtoken.rs

```diff
@@ -129,33 +131,167 @@ fn instant_from_epoch_seconds(secs: u64) -> Option<Instant> {
     }
 }
 
+/// Verifier is used to verify OIDC ID Tokens.
+#[derive(Debug)]
+pub(crate) struct Verifier {
+    jwk_client: JwkClient,
+    audience: Option<String>,
+    email: Option<String>,
+    jwks_url: Option<String>,
+    clock_skew: Duration,
+}
+
+impl Default for Verifier {
+    fn default() -> Self {
+        Self {
+            jwk_client: JwkClient::new(),
+            audience: None,
+            email: None,
+            jwks_url: None,
+            clock_skew: Duration::from_secs(10),
+        }
+    }
+}
+
+impl Verifier {
+    /// Sets the audience for the token verification.
+    pub fn with_audience<S: Into<String>>(mut self, audience: S) -> Self {
```

Comment:
It is a bit odd that I can set the audience, but not clear the audience.

Should we separate this to a builder and once built we disable all functions to change the audience, email or clock skew?